// angular-html-parser/dist/index.mjs
import { TagContentType } from "./compiler/src/ml_parser/tags.mjs";
import { getHtmlTagDefinition } from "./compiler/src/ml_parser/html_tags.mjs";
import { ParseLocation, ParseSourceFile, ParseSourceSpan } from "./compiler/src/parse_util.mjs";
import { RecursiveVisitor, visitAll } from "./compiler/src/ml_parser/ast.mjs";
import { HtmlParser } from "./compiler/src/ml_parser/html_parser.mjs";

//#region src/index.ts
let parser = null;
const getParser = () => {
	if (!parser) parser = new HtmlParser();
	return parser;
};
function parse(input, options = {}) {
	const { canSelfClose = false, allowHtmComponentClosingTags = false, isTagNameCaseSensitive = false, getTagContentType, tokenizeAngularBlocks = false, tokenizeAngularLetDeclaration = false, enableAngularSelectorlessSyntax = false } = options;
	return getParser().parse(input, "angular-html-parser", {
		tokenizeExpansionForms: tokenizeAngularBlocks,
		canSelfClose,
		allowHtmComponentClosingTags,
		tokenizeBlocks: tokenizeAngularBlocks,
		tokenizeLet: tokenizeAngularLetDeclaration,
		selectorlessEnabled: enableAngularSelectorlessSyntax
	}, isTagNameCaseSensitive, getTagContentType);
}

//#endregion
export { ParseLocation, ParseSourceFile, ParseSourceSpan, RecursiveVisitor, TagContentType, getHtmlTagDefinition, parse, visitAll };
