// angular-html-parser/dist/compiler/src/ml_parser/tokens.mjs
//#region ../compiler/src/ml_parser/tokens.ts
let TokenType = /* @__PURE__ */ function(TokenType$1) {
	TokenType$1[TokenType$1["TAG_OPEN_START"] = 0] = "TAG_OPEN_START";
	TokenType$1[TokenType$1["TAG_OPEN_END"] = 1] = "TAG_OPEN_END";
	TokenType$1[TokenType$1["TAG_OPEN_END_VOID"] = 2] = "TAG_OPEN_END_VOID";
	TokenType$1[TokenType$1["TAG_CLOSE"] = 3] = "TAG_CLOSE";
	TokenType$1[TokenType$1["INCOMPLETE_TAG_OPEN"] = 4] = "INCOMPLETE_TAG_OPEN";
	TokenType$1[TokenType$1["TEXT"] = 5] = "TEXT";
	TokenType$1[TokenType$1["ESCAPABLE_RAW_TEXT"] = 6] = "ESCAPABLE_RAW_TEXT";
	TokenType$1[TokenType$1["RAW_TEXT"] = 7] = "RAW_TEXT";
	TokenType$1[TokenType$1["INTERPOLATION"] = 8] = "INTERPOLATION";
	TokenType$1[TokenType$1["ENCODED_ENTITY"] = 9] = "ENCODED_ENTITY";
	TokenType$1[TokenType$1["COMMENT_START"] = 10] = "COMMENT_START";
	TokenType$1[TokenType$1["COMMENT_END"] = 11] = "COMMENT_END";
	TokenType$1[TokenType$1["CDATA_START"] = 12] = "CDATA_START";
	TokenType$1[TokenType$1["CDATA_END"] = 13] = "CDATA_END";
	TokenType$1[TokenType$1["ATTR_NAME"] = 14] = "ATTR_NAME";
	TokenType$1[TokenType$1["ATTR_QUOTE"] = 15] = "ATTR_QUOTE";
	TokenType$1[TokenType$1["ATTR_VALUE_TEXT"] = 16] = "ATTR_VALUE_TEXT";
	TokenType$1[TokenType$1["ATTR_VALUE_INTERPOLATION"] = 17] = "ATTR_VALUE_INTERPOLATION";
	TokenType$1[TokenType$1["DOC_TYPE_START"] = 18] = "DOC_TYPE_START";
	TokenType$1[TokenType$1["DOC_TYPE_END"] = 19] = "DOC_TYPE_END";
	TokenType$1[TokenType$1["EXPANSION_FORM_START"] = 20] = "EXPANSION_FORM_START";
	TokenType$1[TokenType$1["EXPANSION_CASE_VALUE"] = 21] = "EXPANSION_CASE_VALUE";
	TokenType$1[TokenType$1["EXPANSION_CASE_EXP_START"] = 22] = "EXPANSION_CASE_EXP_START";
	TokenType$1[TokenType$1["EXPANSION_CASE_EXP_END"] = 23] = "EXPANSION_CASE_EXP_END";
	TokenType$1[TokenType$1["EXPANSION_FORM_END"] = 24] = "EXPANSION_FORM_END";
	TokenType$1[TokenType$1["BLOCK_OPEN_START"] = 25] = "BLOCK_OPEN_START";
	TokenType$1[TokenType$1["BLOCK_OPEN_END"] = 26] = "BLOCK_OPEN_END";
	TokenType$1[TokenType$1["BLOCK_CLOSE"] = 27] = "BLOCK_CLOSE";
	TokenType$1[TokenType$1["BLOCK_PARAMETER"] = 28] = "BLOCK_PARAMETER";
	TokenType$1[TokenType$1["INCOMPLETE_BLOCK_OPEN"] = 29] = "INCOMPLETE_BLOCK_OPEN";
	TokenType$1[TokenType$1["LET_START"] = 30] = "LET_START";
	TokenType$1[TokenType$1["LET_VALUE"] = 31] = "LET_VALUE";
	TokenType$1[TokenType$1["LET_END"] = 32] = "LET_END";
	TokenType$1[TokenType$1["INCOMPLETE_LET"] = 33] = "INCOMPLETE_LET";
	TokenType$1[TokenType$1["COMPONENT_OPEN_START"] = 34] = "COMPONENT_OPEN_START";
	TokenType$1[TokenType$1["COMPONENT_OPEN_END"] = 35] = "COMPONENT_OPEN_END";
	TokenType$1[TokenType$1["COMPONENT_OPEN_END_VOID"] = 36] = "COMPONENT_OPEN_END_VOID";
	TokenType$1[TokenType$1["COMPONENT_CLOSE"] = 37] = "COMPONENT_CLOSE";
	TokenType$1[TokenType$1["INCOMPLETE_COMPONENT_OPEN"] = 38] = "INCOMPLETE_COMPONENT_OPEN";
	TokenType$1[TokenType$1["DIRECTIVE_NAME"] = 39] = "DIRECTIVE_NAME";
	TokenType$1[TokenType$1["DIRECTIVE_OPEN"] = 40] = "DIRECTIVE_OPEN";
	TokenType$1[TokenType$1["DIRECTIVE_CLOSE"] = 41] = "DIRECTIVE_CLOSE";
	TokenType$1[TokenType$1["EOF"] = 42] = "EOF";
	return TokenType$1;
}({});

//#endregion
export { TokenType };
